File: Output/java/medium/MinimumAreaRectangle.java
/**https://leetcode.com/problems/minimum-area-rectangle/ */
//You are given an array of points in the X-Y plane points where points[i] = [xi, yi].
//Return the minimum area of a rectangle formed from these points, with sides parallel to the X and Y axes. If there is not any such rectangle, return 0.


class MinimumAreaRectangle {
    public int minAreaRect(int[][] points) {
        int n = points.length;
        Map<Integer, Set<Integer>> map = new HashMap<>();
        for(int[] p : points){
            map.putIfAbsent(p[0], new HashSet<>());
            map.get(p[0]).add(p[1]);
        }
        int res = Integer.MAX_VALUE;
        for(int i = 0; i < n; i++){
            for(int j = i + 1; j < n; j++){
                int[] p1 = points[i];
                int[] p2 = points[j];
                if(p1[0] == p2[0] || p1[1] == p2[1]){
                    continue;
                }
                if(map.get(p1[0]).contains(p2[1]) && map.get(p2[0]).contains(p1[1])){
                    res = Math.min(res, Math.abs(p1[0] - p2[0]) * Math.abs(p1[1] - p2[1]));
                }
            }
        }
        return res == Integer.MAX_VALUE ? 0 : res;
        
        
        
    }
}

    

File: Tool/PreUtil/src/output/python/medium/MinimumAreaRectangle.py
#https://leetcode.com/problems/minimum-area-rectangle/
#You are given an array of points in the X-Y plane points where points[i] = [xi, yi].
#Return the minimum area of a rectangle formed from these points, with sides parallel to the X and Y axes. If there is not any such rectangle, return 0.


class Solution(object):
    def minAreaRect(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """     
        res = float('inf')
        points = set(map(tuple, points))
        for i in range(len(points)):
            for j in range(i+1, len(points)):
                x1, y1 = points[i]
                x2, y2 = points[j]
                if x1 != x2 and y1 != y2 and (x1, y2) in points and (x2, y1) in points:
                    res = min(res, abs(x1-x2) * abs(y1-y2))
        return res if res != float('inf') else 0

        

    